idaes/core/solvers.py

```diff
@@ -0,0 +1,71 @@
+from pyomo.environ import SolverFactory
+import idaes
+
+
+_use_idaes_config = False
+
+
+class SolverWrapper(object):
+    def __init__(self, name, register=True):
+        self.name = name
+        if name == 'default':
+            self.solver = None
+            doc = "IDAES Configured Default Solver"
+        else:
+            self.solver = SolverFactory.get_class(name)
+            doc = SolverFactory.doc(name)
+        if register:
+            SolverFactory.unregister(name)
+            # Re-register the solver (register is a decorator)
+            SolverFactory.register(name, doc)(self)
+
+    def __call__(self, *args, **kwargs):
+        global _use_idaes_config
+        if self.name == "default":
+            name = idaes.cfg.default_solver
+            solver = SolverFactory.get_class(name)
+        else:
+            name = self.name
+            solver = self.solver
+        if name in idaes.cfg and (_use_idaes_config or name == "default"):
+            for k, v in idaes.cfg[name].items():
+                if k not in kwargs:
+                    kwargs[k] = v
+                elif k == "options":
+                    # options is in ConfigBlock and in kwargs, treat "options"
+                    # special so individual options can have defaults not just
+                    # the whole options block
+                    for opk, opv in v.items():
+                        if opk not in kwargs["options"]:
+                            kwargs["options"][opk] = opv
+        return solver(*args, **kwargs)
+
+
+def use_idaes_solver_configuration_deafults(b=True):
+    """
+    This function enables (or disables if given False as the argument) solvers
```

Comment:
If `b == False`, the wrappers are not actually disabled...

Reply:
My mistake.  I didn't catch that the boolean flag was polled as part of `__call__` above.  One stylistic thing, though: I have a knee-jerk aversion to global variables.  You could get the same behavior without using Globals by making `_use_idaes_config` a class attribute on `SolverWrapper`.